bin/digestif

```diff
@@ -2,4 +2,14 @@
 -- Remove relative directories from package path
 package.path = package.path:gsub("%f[^\0;]%.[^;]*", ""):gsub(";+", ";"):gsub("^;", ""):gsub(";$", "")
 package.cpath = package.cpath:gsub("%f[^\0;]%.[^;]*", ""):gsub(";+", ";"):gsub("^;", ""):gsub(";$", "")
+
+-- this looks like peculiarity of LuaTeX and ConTeXt specifically its script runner
+-- when I run 'texlua <name> $@' it passes the name of the script as the first argument
+
+-- if first element of arg matches the full path of this file name, remove it
+local fullpath = debug.getinfo(1,"S").source:sub(2)
+if arg[1] == fullpath then
+  table.remove(arg, 1)
+end
```

Comment:
Funny, I can't observe the issue you describe here. I've tried `texlua`, `luatex --luaonly` and `mtxrun --execute`, and `fullpath` is always given as `arg[0]`.

What is `arg[0]` in your situation? (And which platform?)

Reply:
This is probably unique to standalone ConTeXt which I have installed on my Macbook. It does not have `texlua` and the suggestion was to use Lua engine from `LuaMetatex` which is accessible through the Lua script `texlua.lua`. This is how my ConTeXt `bin` directory looks like 

```
lrwxr-xr-x  1 evgeniysharapov  staff    10B Jul  3 00:21 context -> luametatex
-rw-r--r--  1 evgeniysharapov  staff   1.3K Jul  3 00:21 context.lua
-rwxr-xr-x  1 evgeniysharapov  staff   3.0M Jul  3 00:21 luametatex
lrwxr-xr-x  1 evgeniysharapov  staff    10B Jul  3 00:21 mtxrun -> luametatex
-rw-r--r--  1 evgeniysharapov  staff   685K Jul  3 00:21 mtxrun.lua
lrwxr-xr-x  1 evgeniysharapov  staff    10B Jul 24 22:29 texlua -> luametatex
lrwxr-xr-x  1 evgeniysharapov  staff    80B Jul 24 22:33 texlua.lua -> /Users/evgeniysharapov/.ConTeXt/tex/texmf-context/scripts/context/lua/texlua.lua
[evgeniysharapov@SharMbp:~/.ConTeXt/tex/texmf-osx-64/bin]
```

When I run `digestif` I get following `arg` table 

```
arg[0] = texlua
arg[1] = /Users/evgeniysharapov/.digestif/bin/digestif
```

Reply:
Okay, this is a bug in `texlua.lua` then. It's a kind of universal expectation that arg[1] is the first argument to the script, no matter how it was invoked. For comparison:
```
$ python -c "import sys; print(sys.argv)" 1 2 3
['-c', '1', '2', '3']
$ python -O -c "import sys; print(sys.argv)" 1 2 3
['-c', '1', '2', '3']
$ python -O -S -c "import sys; print(sys.argv)" 1 2 3
['-c', '1', '2', '3']
```

As you can see, the script doesn't see the previous argument `-O` etc.

But in any case it seems you can use `luametatex --luaonly [lua_file]` and it handles the arguments correctly.